agent/pkg/crane/k8s/deploy_facade.go

```diff
@@ -56,8 +57,8 @@ func NewDeployFacade(params *DeployFacadeParams) *deployFacade {
 	}
 }
 
-func (d *deployFacade) CheckPreConditions() error {
-	if err := d.namespace.deployNamespace(); err != nil {
+func (d *deployFacade) CheckPreConditions(cfg *config.Configuration) error {
```

Comment:
config should be passed to the deployFacade itself, saving some repetition functions `CheckPreConditions`, `PreDeploy` etc.